src/cljs/rems/application.cljs

```diff
@@ -338,37 +338,42 @@
         show-diff (:show-diff edit-application)
-        field-validations (index-by [:field-id] (:validation-errors edit-application))
+        field-validations (index-by [:form-id :field-id] (:validation-errors edit-application))
         attachments (index-by [:attachment/id] (:application/attachments application))
         form-fields-editable? (form-fields-editable? application)
-        readonly? (not form-fields-editable?)]
-    [collapsible/component
-     {:id "application-fields"
-      :title (text :t.form/application)
-      :always
-      [:div
-       (into [:div]
-             (for [fld (get-in application [:application/form :form/fields])
-                   :when (and (field-visible? fld field-values)
-                              (not (:field/private fld)))] ; private fields will have empty value anyway
-               [fields/field (assoc fld
-                                    :on-change #(rf/dispatch [::set-field-value (:field/id fld) %])
-                                    :on-set-attachment #(rf/dispatch [::save-attachment (:field/id fld) %1 %2])
-                                    :on-remove-attachment #(do
-                                                             (rf/dispatch [::set-field-value (:field/id fld) ""])
-                                                             (rf/dispatch [::set-attachment-success (:field/id fld)]))
-                                    :on-toggle-diff #(rf/dispatch [::toggle-diff (:field/id fld)])
-                                    :field/value (get field-values (:field/id fld))
-                                    :field/attachment (when (= :attachment (:field/type fld))
-                                                        (get attachments (parse-int (:field/value fld))))
-                                    :field/previous-attachment (when (= :attachment (:field/type fld))
-                                                                 (when-let [prev (:field/previous-value fld)]
-                                                                   (get attachments (parse-int prev))))
-                                    :success (= attachment-success (:field/id fld))
-                                    :diff (get show-diff (:field/id fld))
-                                    :validation (field-validations (:field/id fld))
-                                    :readonly readonly?
-                                    :app-id (:application/id application))]))]}]))
+        readonly? (not form-fields-editable?)
+        language @(rf/subscribe [:language])]
+    (into [:div]
+          (for [form (:application/forms application)
+                :let [form-id (:form/id form)]]
+            [collapsible/component
+             {:id "application-fields"
+              :class "mb-3"
+              :title (or (get-in form [:form/title language]) (text :t.form/application))
+              :always (into [:div]
+                            (for [field (:form/fields form)
```

Comment:
should this for be factored out as a function? or does it share too many letted values?

Reply:
I think we can have a look at this later. The field abstraction is not working here so well. I'll add some refactoring tasks.

Reply:
sounds good